modules/rostests/apitests/ntdll/NtStartProfile.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Test for NtStartProfile
+ * COPYRIGHT:   Copyright 2021 Thomas Faber (thomas.faber@reactos.org)
+ */
+
+#include "precomp.h"
+#include <versionhelpers.h>
+
+#define SIZEOF_MDL (5 * sizeof(PVOID) + 2 * sizeof(ULONG))
+typedef ULONG_PTR PFN_NUMBER;
+/* Maximum size that can be described by an MDL on 2003 and earlier */
+#define MAX_MDL_BUFFER_SIZE ((MAXUSHORT - SIZEOF_MDL) / sizeof(PFN_NUMBER) * PAGE_SIZE + PAGE_SIZE - 1)
+
+static BOOL IsWow64;
+static KAFFINITY SystemAffinityMask;
+static ULONG DummyBuffer[4096];
+
+/* The "Buffer[Offset]++;" should likely be within 128 bytes of the start
+ * of the function on any architecture we support
+ */
+#define LOOP_FUNCTION_SIZE_SHIFT 7
+#define LOOP_FUNCTION_SIZE (1UL << LOOP_FUNCTION_SIZE_SHIFT)
+C_ASSERT(LOOP_FUNCTION_SIZE == 128);
+typedef void LOOP_FUNCTION(volatile ULONG *, ULONG, ULONG);
+static
+void
+LoopFunction(
+    _Inout_updates_all_(BufferSize) volatile ULONG *Buffer,
+    _In_ ULONG BufferSizeInElements,
+    _In_ ULONG LoopCount)
+{
+    ULONG i;
+    ULONG Offset;
+
+    for (i = 0; i < LoopCount; i++)
+    {
+        for (Offset = 0; Offset < BufferSizeInElements; Offset++)
+        {
+            Buffer[Offset]++;
```

Comment:
`Buffer` is a pointer-to-volatile. That should be enough to ensure these writes will always occur, as far as I'm aware.

The contents don't get checked anywhere, since the only thing we care about is spending some CPU time here. Some random memory operations seemed like the best way, but happy to hear better suggestions.

I'll need to write some assembly functions for the edge case scenarios, but it seemed like a good idea to have a portable C function to profile, too.